### PR TITLE
Remove unneeded parameters.

### PR DESCRIPTION
**Pull request by bill-dall**
Improve error handling around authentication race conditions.
Improve version checking handling to handle SDK clean install case.

Three things that I ran into yesterday...

### CODE REVIEW

File: Deploy-CAM.ps1 [hunk 1]
```diff
@@ -4,9 +4,7 @@
 
 
 param(
-    $subscriptionId,
```
Comment (bill-dall) [inline Deploy-CAM.ps1]: As @cptera pointed out, these two don't do anything...

File: Deploy-CAM.ps1 [hunk 2]
```diff
@@ -1499,11 +1497,12 @@ function New-ConnectionServiceDeployment() {
                 -Verbose
         }
         else {
-            for($idx = 10;$idx -gt 0;$idx--)
+            $maxRetries = 30
```
Comment (bill-dall) [inline Deploy-CAM.ps1]: I hit this a lot yesterday, sometimes requiring 15 retries. I bumped the threshold to 30.

File: Deploy-CAM.ps1 [hunk 3]
```diff
@@ -1499,11 +1497,12 @@ function New-ConnectionServiceDeployment() {
...
+            for($idx = 0;$idx -lt $maxRetries;$idx++)
             {
                 try {
                     New-AzureRmResourceGroupDeployment `
-                        -DeploymentName "CS" `
+                        -DeploymentName "CS$connectionServiceNumber-$idx" `
```
Comment (bill-dall) [inline Deploy-CAM.ps1]: If the deployment name is not unique then the deployment that was going to succeed will fail.

File: Deploy-CAM.ps1 [hunk 4]
```diff
@@ -2261,19 +2267,43 @@ function Deploy-CAM() {
     }
 }
 
-function Confirm-ModuleVersion {
+function Confirm-ModuleVersion()
```
Comment (bill-dall) [inline Deploy-CAM.ps1]: Confirm-ModuleVersion was giving me errors on my laptop. The reason was or seems to be that AzureRM as a module is not installed with the newest SDK installer, just a new version of Azure module (and all the AzureRM sub-modules...) so this seems to be a valid configuration...

File: Deploy-CAM.ps1 [hunk 5]
```diff
@@ -2261,19 +2267,43 @@ function Deploy-CAM() {
...
+            Write-Host ("AzureRM module version must be equal or greater than " + $MinAzureRMVersion)
+            return $false
+        }
     }
+    else {
+        # the Azure SDK doesn't install 'AzureRM' as a base module any more, just Azure
+        $MinAzureVersion="5.0.0"
+        $AzureModule = Get-Module -ListAvailable -Name "Azure"
+
+        if ( $AzureModule ) {
```
[the remark this answers is not in the document]
Comment (bill-dall) [reply Deploy-CAM.ps1]: Good point!

File: Deploy-CAM.ps1 [hunk 6]
```diff
@@ -2261,19 +2267,45 @@ function Deploy-CAM() {
...
+            return $false
+        }
+    }
+    else {
+        # the Azure SDK doesn't install 'AzureRM' as a base module any more, just Azure
+        $MinAzureVersion="5.0.0"
+        $AzureModule = Get-Module -ListAvailable -Name "Azure"
+
+        if ( -not $AzureModule ) {
+            # neither module found
```
Comment (jackyycheng) [inline Deploy-CAM.ps1]: Would it be better to log something here so user know what to do?

**Pull request merged**